skimage/filters/thresholding.py

```diff
@@ -881,3 +881,42 @@ def threshold_sauvola(image, window_size=15, k=0.2, r=None):
         r = 0.5 * (imax - imin)
     m, s = _mean_std(image, window_size)
     return m * (1 + k * ((s / r) - 1))
+
+
+def apply_hysteresis_threshold(image, low, high):
+    """Apply hysteresis thresholding to `image`.
+
+    This algorithm finds regions where `image` is greater than `high`
+    OR `image` is greater than `low` *and* that region is connected to
+    a region greater than `high`.
+
+    Parameters
+    ----------
+    image : array
+        The input image.
+    low : float
+        The low threshold.
+    high : float
+        The high threshold.
+
+    Returns
+    -------
+    thresholded : array of bool, same shape as `image`
+        Array in which `True` indicates the locations where `image`
+        was above the hysteresis threshold.
+
```

Comment:
an appropriate reference to add would be:
J. Canny.  A computational approach to edge detection.  IEEE Transactions on Pattern Analysis and Machine Intelligence. 1986; vol. 8, pp.679-698.  DOI: 10.1109/TPAMI.1986.4767851